helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -573,35 +596,31 @@ public void resetPartition(String clusterName, String instanceName, String resou
     // check the instance is alive
     LiveInstance liveInstance = accessor.getProperty(keyBuilder.liveInstance(instanceName));
     if (liveInstance == null) {
-      throw new HelixException(
-          "Can't reset state for " + resourceName + "/" + partitionNames + " on " + instanceName
-              + ", because " + instanceName + " is not alive");
+      // check if the instance has added to the cluster
+      String instanceConfigPath = PropertyPathBuilder.instanceConfig(clusterName, instanceName);
+      throw new HelixException(String.format(
+          _zkClient.exists(instanceConfigPath) ? ResetPartitionFailureReason.INSTANCE_NOTALIVE
+              .getMessage(instanceName, resourceName, partitionNames)
+              : ResetPartitionFailureReason.INSTANCE_NONEXISTENT
+                  .getMessage(instanceName, resourceName, partitionNames), instanceName,
+          clusterName));
```

Comment:
I would suggest passing the cluster name to getMessage, and let it return the fully formated string.

Reply:
Some messages require 2 args an others need one arg. Please correct me if I am wrong but I think it is hard to return a fully formated string if we do not add extra logic here.